Dockerfile

```diff
@@ -0,0 +1,58 @@
+
+# Accept the Go version for the image to be set as a build argument.
+# Default to Go 1.11
+ARG GO_VERSION=1.11
+ARG GOARCH=amd64
+
+# First stage: build the executable.
+FROM golang:${GO_VERSION}-alpine AS builder
+
+# Create the user and group files that will be used in the running container to
+# run the process as an unprivileged user.
+RUN mkdir /user && \
+    echo 'nobody:x:65534:65534:nobody:/:' > /user/passwd && \
```

Comment:
Is it temporary?

Reply:
It shouldn't be, since the artifacts are copied over in the second stage. 

Realistically this could be a lot simpler, and while writing this I think I'm convinced to clean it up.